src/syrupy/report.py

```diff
@@ -198,8 +198,14 @@ def unused(self) -> "SnapshotFossils":
                 unused_snapshots = {
                     snapshot
                     for snapshot in unused_snapshot_fossil
-                    if self._selected_items_match_name(snapshot.name)
-                    or self._provided_nodes_match_name(snapshot.name, provided_nodes)
+                    if self._selected_items_match_name(
+                        snapshot_location=snapshot_location, snapshot_name=snapshot.name
+                    )
+                    and self._provided_nodes_match_name(
+                        snapshot_location=snapshot_location,
+                        snapshot_name=snapshot.name,
+                        provided_nodes=provided_nodes,
```

Comment:
This bit might actually be dead code. It's never hit in tests. I don't think it's harmful at the moment, so will loop back to see if it can be removed, or if there's some edge case we're not testing.